v2/pkg/genruntime/secrets/errors.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package secrets
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// SecretNotOwnedError indicates the secret is not owned by the resource attempting to write it
+type SecretNotOwnedError struct {
+	Namespace  string
+	SecretName string
+	ObjectName string
+}
+
+var _ error = &SecretNotOwnedError{}
+
+func AsSecretNotOwnedError(err error) (*SecretNotOwnedError, bool) {
```

Comment:
Does this need to deal with wrapped errors too? I think there's a standard `Unwrap()` method you can check for.

Reply:
`errors.As` calls that internally, so this already deals with wrapped errors.